coreutils/echo.rs

```diff
@@ -1,651 +1,21 @@
-use crate::libbb::ptr_to_globals::bb_errno;
-use crate::libbb::xfuncs_printf::xmalloc;
-use crate::librb::size_t;
-use libc;
-use libc::free;
-extern "C" {
+use std::env;
 
-  #[no_mangle]
-  fn strlen(__s: *const libc::c_char) -> size_t;
-  #[no_mangle]
-  fn stpcpy(_: *mut libc::c_char, _: *const libc::c_char) -> *mut libc::c_char;
+pub fn echo_main() {
+    let mut output = String::new();
+    let args: Vec<String> = env::args().collect();
```

Comment:
Instead of doing this manual args parsing, we should probably use [SafeStyle](https://github.com/samuela/rustybox/blob/master/applets/applet_tables.rs#L35) instead. Here's an example: https://github.com/samuela/rustybox/blob/master/applets/applet_tables.rs#L2782 and https://github.com/samuela/rustybox/blob/master/coreutils/true.rs.

Reply:
There are also a few different args parsing libraries in rust. Thus far rustybox hasn't really settled on one, but we should probably pick one here and stick with it.

Reply:
@samuela Regarding the question of which argument parser to go with, [clap](https://github.com/clap-rs/clap)'s author has given some suggestions in this Reddit comment: https://www.reddit.com/r/rust/comments/8i5k3l/looking_for_a_tiniest_args_parser/dyp8l13?utm_source=share&utm_medium=web2x. Clap also looks like a good choice to go with.

Reply:
I've used getopts before, and it seems to be the de facto choice for simple arg parsing, so I say we go with that. clap is a great framework, but it is fairly heavyweight. Clap weighs in at a surprising [202kb](https://crates.io/crates/clap), whereas getopts is only [18kb](https://crates.io/crates/getopts).

Reply:
FYI `getopts` only works for UTF-8 strings, so you can’t process non-UTF-8 filenames with it.  It looks like `SafeStyle` only accepts `&str` arguments though anyway.

Reply:
@Arcterus Oh interesting, I was not aware of that. Has that come up as an issue at all for https://github.com/uutils/coreutils?

Reply:
Yeah, we are trying to convert everything to `clap` in part because of this issue.

Reply:
Ah, that's unfortunate... I don't think we really need the entirety of clap and I'd like to keep things as small as possible since that's a major selling point for rustybox/busybox.

Reply:
Ok, I created https://github.com/rust-lang/getopts/issues/97!

Reply:
@samuela Just saw the updates on the thread. I am stuck somewhere, plus, I have some questions about the build and test parts of the project that would not be suitable to ask here. Do you have a way I could contact separately like a Slack channel, Microsoft Teams channel, email thread or something else?

Reply:
Sure, you can contact me over email at the email listed in my github account, and I'd be happy to answer any questions

Reply:
Hey @samuela sent you an email! Could you find some time to look at it? Since it's not from Gmail, it might be in your spam.

Reply:
https://github.com/rust-lang/getopts/issues/97#issuecomment-653632408

argh is quite small, and there's some traction there to support non-utf8.

There's also pico-args, which to my knowledge is the smallest out there. And it also supports non-utf8. Main downside being the help text isn't autogenerated.

Related discussion [here](https://www.reddit.com/r/rust/comments/hjn64e/dua_28_disk_usage_analyzer_is_now_430kb_smaller/fwnxh9b/).

Reply:
Thanks for the pointers @joshuarli ! I wasn't aware of the argh project. That may be a suitable alternative.